Jenkinsfile

```diff
@@ -20,5 +22,42 @@ pipeline {
                 sh label: 'Running Packer build', script: 'packer build pipeline/packer/mhs.json'
             }
         }
+
+        stage('Deploy') {
+            steps {
+                dir('pipeline/terraform/test-environment') {
+                    sh label: 'Initialising Terraform', script: 'terraform init -input=false'
+                    sh label: 'Applying Terraform configuration', script: """
+                            terraform apply -auto-approve \
+                            --var cluster_id=${CLUSTER_ID} \
+                            -var ecs_address=${DOCKER_REPOSITORY} \
+                            -var task_execution_role=${TASK_EXECUTION_ROLE} \
+                            -var build_id=${BUILD_TAG}
+                        """
+                }
+            }
+        }
+
+        stage('Integration Tests') {
+            steps {
+                // Wait for MHS container to fully stand up
+                // TODO: Run actual integration tests.
+                sh label: 'Ping MHS', script: 'sleep 20; curl ${MHS_ADDRESS}'
```

Comment:
Yeah I tried a few different things with Jenkin's `waitUntil` but it got a little messy, since jenkins steps will fail completely if the script fails anyway (curl will throw an exit code 7 if port 80 is refused, causing the step to fail). 

I probably could have wrapped the script in try catch but for the time being I think this will suffice, I think we'll have to modify this if we stand up multiple services (caches / queues /different components) but it works for the time being